src/main/java/gyro/google/compute/InstanceTemplateResource.java

```diff
@@ -0,0 +1,272 @@
+import gyro.core.scope.State;
+import gyro.core.validation.ConflictsWith;
+import gyro.core.validation.DependsOn;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidationError;
+import gyro.google.Copyable;
+
+/**
+ * Creates an Instance Template.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     google::compute-instance-template instance-template-example
+ *         name: "instance-template-example"
+ *         description: "Instance template example"
+ *
+ *         properties
+ *             disk
+ *                 auto-delete: true
+ *                 boot: true
+ *
+ *                 initialize-params
+ *                     disk-name: "instance-template-example-disk-1"
+ *                     source-image: "projects/debian-cloud/global/images/family/debian-9"
+ *                 end
+ *             end
+ *
+ *             disk
+ *                 auto-delete: true
+ *                 boot: false
+ *
+ *                 initialize-params
+ *                     disk-name: "instance-template-example-disk-2"
+ *                     source-image: "projects/debian-cloud/global/images/family/debian-9"
+ *                 end
+ *             end
+ *
+ *             machine-type: "f1-micro"
+ *
+ *             network-interface
+ *                 network: $(external-query google::compute-network {name: "default"})
+ *
+ *                 access-config
+ *                     name: "External NAT"
+ *                     type: "ONE_TO_ONE_NAT"
+ *                     network-tier: "PREMIUM"
+ *                 end
+ *             end
+ *
+ *             scheduling
+ *                 automatic-restart: true
+ *                 on-host-maintenance: "MIGRATE"
+ *                 preemptible: false
+ *             end
+ *         end
+ *     end
+ */
+@Type("compute-instance-template")
+public class InstanceTemplateResource extends ComputeResource implements Copyable<InstanceTemplate> {
+
+    private String description;
+
+    private String name;
+
+    private ComputeInstanceProperties properties;
+
+    private String selfLink;
+
+    private InstanceResource sourceInstance;
+
+    private ComputeSourceInstanceParams sourceInstanceParams;
+
+    /**
+     * Description of this resource.
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    /**
+     * Name of the resource.
+     * The name must be 1-63 characters long, and comply with RFC1035.
+     * Specifically, the name must be 1-63 characters long and match the regular expression ``[a-z]([-a-z0-9]*[a-z0-9])?``
+     * which means the first character must be a lowercase letter, and all following characters must be a dash,
+     * lowercase letter, or digit, except the last character, which cannot be a dash.
+     */
+    @Required
+    @Regex("[a-z]([-a-z0-9]*[a-z0-9])?")
```

Comment:
The regex does not check for the character limitation.
`"[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?"` is more close to what it should be.